--- a/engine/vcl/source/gdi/virdev.cxx
+++ b/engine/vcl/source/gdi/virdev.cxx
@@ -380,13 +380,19 @@
     assert(meFormatAndAlpha == DeviceFormat::WITHOUT_ALPHA);
     assert(mpVirDev);
     assert( rNewSize != GetOutputSizePixel() &&  "Trying to re-use a VirtualDevice but this time using a pre-allocated buffer");
     assert( rNewSize.Width() >= 1 );
     assert( rNewSize.Height() >= 1 );
 
+#ifdef IOS
+    // iOS doesn't support SetSizeUsingBuffer. Not setting the size here means
+    // that no visible content will render to slides in presentations, for instance...
+    bool bRet = mpVirDev->SetSize( rNewSize.Width(), rNewSize.Height(), false );
+#else
     bool bRet = mpVirDev->SetSizeUsingBuffer( rNewSize.Width(), rNewSize.Height(), pBuffer );
+#endif
     if ( bRet )
     {
         mnOutWidth  = rNewSize.Width();
         mnOutHeight = rNewSize.Height();
     }
 
